### PR TITLE
Create Team modal

### PR DESCRIPTION
**Pull request by tiagoevanp**

### CODE REVIEW

File: client/sidebar/header/CreateTeam.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+		}
+		const regex = new RegExp(`^${ namesValidation }$`);
+
+		return regex;
+	}, [allowSpecialNames, namesValidation]);
+
+	const [nameError, setNameError] = useState();
+
+	const checkName = useDebouncedCallback(async (name) => {
+		setNameError(false);
```
Comment (tassoevan) [inline client/sidebar/header/CreateTeam.js:43]: Use a single type! `string` or `boolean`, choose one and stick with it.

File: client/sidebar/header/actions/CreateRoomList.js [hunk 2]
```diff
@@ -53,10 +56,12 @@ function CreateRoomList() {
 	const setModal = useSetModal();
 
 	const canCreateChannel = useAtLeastOnePermission(CREATE_CHANNEL_PERMISSIONS);
+	const canCreateTeam = useAtLeastOnePermission(CREATE_TEAM_PERMISSIONS);
 	const canCreateDirectMessages = usePermission('create-d');
 	const canCreateDiscussion = useAtLeastOnePermission(CREATE_DISCUSSION_PERMISSIONS);
 
 	const createChannel = useReactModal(setModal, CreateChannel);
+	const createTeam = useReactModal(setModal, CreateTeam);
```
Comment (tassoevan) [inline client/sidebar/header/actions/CreateRoomList.js:64]: Why not refactor `useReactModal` like here?

https://github.com/RocketChat/Rocket.Chat/pull/21080/files#diff-c15dcafcf94ea74e0e9e57364940d7c475da0dd26fb28a22fa482ec8374f890cR39-R55

There will be performance issue that justifies passing `setModal` as injected dependency.

File: client/sidebar/header/actions/CreateRoomList.js [hunk 3]
```diff
@@ -69,6 +74,7 @@ function CreateRoomList() {
 		<ul className='rc-popover__list'>
 			<Margins block='x8'>
 				{canCreateChannel && <CreateRoomListItem icon={'hashtag'} text={t('Channel')} action={createChannel} />}
+				{canCreateTeam && <CreateRoomListItem icon={'team'} text={t('Team')} action={createTeam} />}
```
Comment (tassoevan) [inline client/sidebar/header/actions/CreateRoomList.js:77]:
```suggestion
				{canCreateTeam && <CreateRoomListItem icon='team' text={t('Team')} action={createTeam} />}
```

**Review by tassoevan — CHANGES_REQUESTED** [review]
Why not defining `CreateTeam` as a TypeScript module?

File: client/sidebar/header/actions/CreateRoomList.js (hunk 2 above)
Comment (tiagoevanp) [reply client/sidebar/header/actions/CreateRoomList.js:64]: I was waiting for that merge situation...

File: client/sidebar/header/CreateTeam.js (hunk 1 above)
Comment (tiagoevanp) [reply client/sidebar/header/CreateTeam.js]: Ok!

File: client/sidebar/header/actions/CreateRoomList.js (hunk 2 above)
Comment (tiagoevanp) [reply client/sidebar/header/actions/CreateRoomList.js:64]: ... well, I will include that in here!

**Review by tassoevan — APPROVED** [review]

**Pull request merged**